dtrace-provider.js

```diff
@@ -10,33 +10,32 @@ DTraceProviderStub.prototype.enable = function() {};
 DTraceProviderStub.prototype.fire = function() {};
 DTraceProviderStub.prototype.disable = function() {};
 
+var reqmode = process.env.NODE_DTRACE_PROVIDER_REQUIRE;
 var builds = ['Release', 'default', 'Debug'];
 var err = null;
 
-for (var i in builds) {
-    try {
-        var binding = require('./build/' + builds[i] + '/DTraceProviderBindings');
-        DTraceProvider = binding.DTraceProvider;
-        break;
-    } catch (e) {
-        if (err === null) {
-            err = e;
+if (reqmode === 'skip') {
+    DTraceProvider = DTraceProviderStub;
+} else {
+    for (var i = 0; i < builds.length; i++) {
+        try {
+            var binding = require('./src/build/' + builds[i] + '/DTraceProviderBindings');
+            DTraceProvider = binding.DTraceProvider;
+            break;
+        } catch (e) {
+            if (err === null) {
+                err = e;
+            }
         }
     }
-}
 
-if (!DTraceProvider) {
-    /*
-     * If the platform looks like it _should_ have DTrace
-     * available, log a failure to load the bindings.
-     */
-    if (process.platform === 'darwin' ||
-        process.platform === 'sunos' ||
-        process.platform === 'freebsd') {
-        console.error(err);
+    if (!DTraceProvider) {
+        if (reqmode === 'hard') {
+            throw err;
```

Comment:
Do we consider the error message part of the API of this module (e.g. if there are users parsing that error message in CI environments because it's such a common error)? If so, does throwing here instead of using `console.log` change the output?

Reply:
No, the error message isn't part of the API, and the last time I changed it (from printing three times to printing once), it didn't seem to affect anyone.

In order for anyone to get the thrown error, they'll need to set NODE_DTRACE_PROVIDER_REQUIRE=hard in their environment, so this shouldn't change anyone's current behaviour.

Reply:
Sounds good to me.